projects/optic/src/commands/diff/diff-all.ts

```diff
@@ -0,0 +1,452 @@
+        path: candidate.to!,
+      });
+      continue;
+    }
+  }
+  return {
+    warnings,
+    results,
+  };
+}
+
+type Result = Awaited<ReturnType<typeof compute>> & {
+  fromParseResults: ParseResult;
+  toParseResults: ParseResult;
+  from?: string;
+  to?: string;
+};
+
+type Warnings = {
+  missingOpticUrl: {
+    path: string;
+  }[];
+  unparseableFromSpec: {
+    path: string;
+    error: unknown;
+  }[];
+  unparseableToSpec: { path: string; error: unknown }[];
+};
+
+function handleWarnings(warnings: Warnings, options: DiffAllActionOptions) {
+  if (warnings.missingOpticUrl.length > 0) {
+    logger.info(
+      chalk.yellow(
+        'Warning - the following OpenAPI specs were detected but did not have x-optic-url keys. `optic diff-all` only runs on specs that include `x-optic-url` keys.'
+      )
+    );
+    logger.info('Run the `optic api add` command to add these specs to optic');
+    logger.info(warnings.missingOpticUrl.map((f) => f.path).join('\n'));
+    logger.info('');
+  }
+
+  if (warnings.unparseableFromSpec.length > 0) {
+    logger.error(
+      chalk.red(
+        `Error - the following specs could not be parsed from the ref ${options.compareFrom}`
+      )
+    );
+
+    for (const unparseableFrom of warnings.unparseableFromSpec) {
+      logger.error(`spec: ${unparseableFrom.path}`);
+      logger.error(unparseableFrom.error);
+      logger.error('');
+    }
+  }
+
+  if (warnings.unparseableToSpec.length > 0) {
+    logger.error(
+      chalk.red(
+        `Error - the following specs could not be parsed from the ${
+          options.compareTo
+            ? `ref ${options.compareTo}`
+            : 'current working directory'
+        }`
+      )
+    );
+
+    for (const unparseableTo of warnings.unparseableToSpec) {
+      logger.error(`spec: ${unparseableTo.path}`);
+      logger.error(unparseableTo.error);
+      logger.error('');
+    }
+    process.exitCode = 1;
+  }
+}
+
+async function openWebpage(
+  { fromParseResults, toParseResults, specResults }: Result,
+  config: OpticCliConfig
+) {
+  const meta = {
+    createdAt: new Date(),
+    command: ['optic', ...process.argv.slice(2)].join(' '),
+  };
+
+  const compressedData = compressData(
+    fromParseResults,
+    toParseResults,
+    specResults,
+    meta
+  );
+  const anonymousId = await getAnonId();
+  trackEvent('optic.diff_all.view_web', anonymousId, {
+    compressedDataLength: compressedData.length,
+  });
+  await open(`${config.client.getWebBase()}/cli/diff#${compressedData}`, {
+    wait: false,
+  });
+}
+
+const getDiffAllAction =
+  (config: OpticCliConfig) => async (options: DiffAllActionOptions) => {
+    if (config.vcs?.type !== VCS.Git) {
+      logger.error(
+        `Error: optic diff-all must be called from a git repository.`
+      );
+      process.exitCode = 1;
+      return;
+    }
+
+    if (options.json) {
+      // For json output we only want to render json
+      logger.setLevel('silent');
+    }
```

Comment:
If there's a better way to do this we can - i also thoguht maybe we could just dump a json output file as well

Reply:
I think this is fine for now. We'll def. hear about it if it's an issue.